Build fails on REST endpoint when `@BeanParam` has parameterized type

### Describe the bug

With Quarkus 2.7.4.Final (and earlier versions), defining a REST endpoint with RestEasy Reactive where a `@BeanParam` argument has a parameterized type, the build fails with `IndexOutOfBoundsException`.

```java
@GET
@Path("topics")
@Produces(MediaType.APPLICATION_JSON)
@Operation(operationId = Operations.GET_TOPICS_LIST)
@APIResponseSchema(Types.TopicList.class)
CompletionStage<Response> listTopics(@QueryParam("filter") String filter,
                                     @BeanParam OrderByInput<TopicOrderKey> orderParams)
```

```java
class OrderByInput<T> {
    @QueryParam("orderKey")
    T field;

    @QueryParam("order")
    String order;
}
```

```java
    @Schema(name = "TopicOrderKey",
            enumeration = { "name", "partitions", "retention.ms", "retention.bytes" })
    public enum TopicOrderKey {
        NAME("name"),
        PARTITIONS("partitions"),
        RETENTION_MS("retention.ms"),
        RETENTION_BYTES("retention.bytes");

        String value;

        private TopicOrderKey(String value) {
            this.value = value;
        }

        public String getValue() {
            return value;
        }

        public static TopicOrderKey fromString(String value) {
            return Arrays.stream(values())
                    .filter(v -> v.getValue().equals(value))
                    .findFirst()
                    .orElse(null);
        }
    }
```

### Expected behavior

Type parameter declared on method argument is applied to the bean parameter object.

### Actual behavior

Stack trace with root cause:
```
Caused by: java.lang.IndexOutOfBoundsException: Index: 0
	at java.base/java.util.Collections$EmptyList.get(Collections.java:4586)
	at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.resolveTypeVariable(EndpointIndexer.java:910)
	at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.toClassName(EndpointIndexer.java:886)
	at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.extractParameterInfo(EndpointIndexer.java:1151)
	at org.jboss.resteasy.reactive.server.processor.ServerEndpointIndexer.scanInjectableBean(ServerEndpointIndexer.java:246)
	at org.jboss.resteasy.reactive.server.processor.ServerEndpointIndexer.handleBeanParam(ServerEndpointIndexer.java:167)
	at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:523)
	... 15 more
```

### How to Reproduce?

Define a REST endpoint method with query parameters defined within a POJO having a generic type parameter where the parameter is supplied in a method parameter annotated with `@BeanParam`.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)

### Additional information

It looks as though a parameterized type passed to `ServerEndpointIndexer#handleBeanParam` as `paramType` is needed to resolve type variables within `EndpointIndexer#resolveTypeVariable`

--- a/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java
+++ b/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java
@@ -1037,17 +1037,41 @@ public abstract class EndpointIndexer<T extends EndpointIndexer<T, PARAM, METHOD
             }
         }
         if (pos != -1) {
-            List<Type> params = JandexUtil.resolveTypeParameters(actualEndpointClass.name(), currentClass.name(),
-                    indexView);
-
-            Type resolved = params.get(pos);
-            if (resolved.kind() != Type.Kind.TYPE_VARIABLE
-                    || !resolved.asTypeVariable().identifier().equals(typeVariable.identifier())) {
+            Type resolved = resolveTypeParameters(pos, actualEndpointClass, currentClass, indexView);
+            if (resolved != null
+                    && (resolved.kind() != Type.Kind.TYPE_VARIABLE
+                            || !resolved.asTypeVariable().identifier().equals(typeVariable.identifier()))) {
                 return resolved;
             }
         }
 
         return typeVariable.bounds().get(0);
+    }
+
+    private static Type resolveTypeParameters(int position, ClassInfo actualEndpointClass, ClassInfo currentClass,
+            IndexView indexView) {
+        List<Type> params = JandexUtil.resolveTypeParameters(actualEndpointClass.name(), currentClass.name(),
+                indexView);
+
+        if (position < params.size()) {
+            return params.get(position);
+        }
+
+        // if not found, try to find the type from the methods
+        for (MethodInfo method : actualEndpointClass.methods()) {
+            for (int paramIndex = 0; paramIndex < method.parametersCount(); paramIndex++) {
+                Type paramType = method.parameterType(paramIndex);
+                if (paramType.kind() == Kind.PARAMETERIZED_TYPE) {
+                    ParameterizedType parameterizedType = paramType.asParameterizedType();
+                    if (parameterizedType.name().equals(currentClass.name())
+                            && position < parameterizedType.arguments().size()) {
+                        return parameterizedType.arguments().get(position);
+                    }
+                }
+            }
+        }
+
+        return null;
     }
 
     private static String appendPath(String prefix, String suffix) {
@@ -1265,7 +1289,7 @@ public abstract class EndpointIndexer<T extends EndpointIndexer<T, PARAM, METHOD
                 typeHandled = true;
                 elementType = toClassName(pt, currentClassInfo, actualEndpointInfo, index);
                 handleOtherParam(existingConverters, errorLocation, hasRuntimeConverters, builder, elementType);
-            } else {
+            } else if (builder.getType() != ParameterType.BEAN) {
                 // the "element" type is not of importance as in this case the signature is used at runtime to determine the proper types
                 elementType = DUMMY_ELEMENT_TYPE.toString();
                 addReaderForType(additionalReaders, pt);

--- a/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/resource/basic/ResourceLocatorTest.java
+++ b/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/resource/basic/ResourceLocatorTest.java
@@ -152,4 +152,12 @@ public class ResourceLocatorTest {
                 .then()
                 .body(is("first and second"));
     }
+
+    @Test
+    @DisplayName("Test @BeanParam annotation using generics in Subresources")
+    public void testBeanParamsWithGenericsInSubresource() {
+        given().get("/sub3/first/resources/subresource4?value=second")
+                .then()
+                .body(is("first and second"));
+    }
 }

--- a/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/resource/basic/resource/ResourceLocatorSubresource.java
+++ b/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/resource/basic/resource/ResourceLocatorSubresource.java
@@ -60,10 +60,23 @@ public class ResourceLocatorSubresource {
         return params.param + " and " + params.value;
     }
 
+    @GET
+    @Path("/subresource4")
+    public String getValueFromGenericBeanParam(@BeanParam ParamsWithGeneric<String> params) {
+        return params.param + " and " + params.value;
+    }
+
     public static class Params {
         @RestPath
         String param;
         @QueryParam("value")
         String value;
+    }
+
+    public static class ParamsWithGeneric<T> {
+        @RestPath
+        String param;
+        @QueryParam("value")
+        T value;
     }
 }

